### PR TITLE
Dev/z.abramowska/paser

### PR DESCRIPTION
**Pull request by zos**
Refactoring and cleanup

* remove openssl from veraison-verifier reqwest dependency
* refactor functionality of realm-verifier JSON parser
* cleanup implementation of realm-verifier Verifier

### CODE REVIEW

**Comment by p-sawicki2** [comment]
Do you plan to support multiple metadata files describing realms and their measurements? I mean the scenario, where one instance of the image-registry is accessed from different realms, each having a different RIM.

File: lib/realm-verifier/src/parser/json.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    pub rems: Vec<Vec<String>>,
+    #[serde(rename = "hash-algo")]
+    pub hash_algo: String,
+}
+
+pub const REFERENCE_RIM: &'static str = "rim";
+pub const REFERENCE_REMS: &'static str = "rems";
+pub const REFERENCE_HASH_ALGO: &'static str = "hash-algo";
+
+impl TryFrom<ReferenceValues> for crate::RealmMeasurements {
```
Comment (Havner) [inline lib/realm-verifier/src/parser/json.rs]: crate:: unneeded

File: lib/realm-verifier/src/parser/json.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        Ok(Self {
+            initial: rim,
+            extensible: rems,
+            hash_algo: crate::HashAlgo { name: hash_algo.clone(), len: required_len },
+        })
+    }
+}
+
+pub fn parse_value(value: serde_json::Value) -> Result<crate::RealmMeasurements, &'static str> {
```
Comment (Havner) [inline lib/realm-verifier/src/parser/json.rs]: crate:: unneeded

File: lib/realm-verifier/src/lib.rs [hunk 3]
```diff
@@ -7,55 +7,99 @@ use tinyvec::ArrayVec;
 
 pub const MAX_MEASUREMENT_SIZE: usize = 64;
 
-pub type MeasurementValue = ArrayVec<[u8; MAX_MEASUREMENT_SIZE]>;
+#[derive(Debug, Clone, Copy)]
```
Comment (Havner) [inline lib/realm-verifier/src/lib.rs:14]: Remove Copy unless it's absolutely needed by some external trait bounds.

File: lib/realm-verifier/src/lib.rs [hunk 4]
```diff
@@ -7,55 +7,99 @@ use tinyvec::ArrayVec;
...
+        self.value.as_slice()
+    }
+
+    pub fn as_mut_slice(&mut self) -> &mut[u8] {
+        self.value.as_mut_slice()
+    }
+}
+
+#[derive(Debug)]
+pub struct HashAlgo {
```
Comment (Havner) [inline lib/realm-verifier/src/lib.rs]: This struct should be enum. In its current form you can create inconsistent values.

File: lib/realm-verifier/src/lib.rs [hunk 5]
```diff
@@ -7,55 +7,99 @@ use tinyvec::ArrayVec;
...
     pub extensible: Vec<[MeasurementValue; CLAIM_COUNT_REALM_EXTENSIBLE_MEASUREMENTS]>,
+    pub hash_algo: HashAlgo,
 }
 
 #[derive(Debug)]
 pub struct RealmVerifier {
     reference_measurements: RealmMeasurements,
 }
 
+macro_rules! eq_msg {
```
Comment (Havner) [inline lib/realm-verifier/src/lib.rs]: Would be nicer to be a template (PartialEq) function. I don't think macro is needed here.

**Review by Havner — CHANGES_REQUESTED** [review]
Some small cosmetics

**Comment by zos** [comment]
> Do you plan to support multiple metadata files describing realms and their measurements? I mean the scenario, where one instance of the image-registry is accessed from different realms, each having a different RIM.

What would identify a Realm to realm-verifier? Currently all realm-verifier knows about the Realm is what is given through token passed by ra-tls.

File: lib/realm-verifier/src/parser/json.rs (hunk 1 above)
Comment (zos) [reply lib/realm-verifier/src/parser/json.rs]: done

File: lib/realm-verifier/src/parser/json.rs (hunk 2 above)
Comment (zos) [reply lib/realm-verifier/src/parser/json.rs]: done

File: lib/realm-verifier/src/lib.rs (hunk 4 above)
Comment (zos) [reply lib/realm-verifier/src/lib.rs]: Not enum, but fixed.

File: lib/realm-verifier/src/lib.rs (hunk 5 above)
Comment (zos) [reply lib/realm-verifier/src/lib.rs]: done

File: lib/realm-verifier/src/lib.rs (hunk 3 above)
Comment (zos) [reply lib/realm-verifier/src/lib.rs:14]: It is required by lib/realm-verifier/src/parser/json.rs:46:17

**Comment by p-sawicki2** [comment]
> What would identify a Realm to realm-verifier? Currently all realm-verifier knows about the Realm is what is given through token passed by ra-tls.

RIM can be used to uniquely identify a Realm. The verifier can keep a collection of reference measurements for different realms (e.g. a map from RIMs to a set of Vec\<REM\> constructed from multiple realm.json files). When the verifier receives a realm token for verification, firstly it needs to find matching RIM within that collection, then check which vector of 4 REMs match the content of the token.

**Pull request merged**